webroot/js/components/chat/chat-message-view.js

```diff
@@ -11,26 +11,39 @@ import { convertToText } from '../../utils/chat.js';
 import { SOCKET_MESSAGE_TYPES } from '../../utils/websocket.js';
 
 export default class ChatMessageView extends Component {
-  async componentDidUpdate(prevProps) {
+  constructor(props) {
+    super(props);
+    this.state = {
+      formattedMessage: '',
+    };
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    const { formattedMessage } = this.state;
+    const { formattedMessage: nextFormattedMessage } = nextState;
+
+    return (formattedMessage !== nextFormattedMessage);
+  }
+
+  async componentDidMount() {
     const { message, username } = this.props;
-    if (prevProps.message === message && this.state.formattedMessage) {
```

Comment:
@mattdsteele I converted this to `componentDidMount` instead!  We also found that rare edge case of formattedMessage coming up empty, which ended up failing out of here, and then calling setState in a crazy loop!